core/domain/src/dds_client.rs

```diff
@@ -0,0 +1,371 @@
+//! A client for talking to the Domain Discovery Service (DDS).
+//!
+//! This module provides:
+//! 1) A clonable `DdsClient` you can share across tasks.
+//! 2) An async `register` method with exponential backoff.
+//! 3) A `shutdown` method to tell DDS you’re going away.
+//!
+//! Uses:
+//! - `Arc<RwLock<…>>` for shared mutable state.
+//! - `tokio::async` + `reqwest` for nonblocking HTTP.
+//! - `thiserror` + `serde` for errors & (de)serialization.
+
+use std::{
+    sync::{Arc},
+    time::{Duration, Instant},
+};
+
+use chrono::Utc;
+use hex::encode as hex_encode;
+use k256::{
+    ecdsa::{signature::Signer, SigningKey},
+    elliptic_curve::sec1::ToEncodedPoint,
+    PublicKey,
+};
+use reqwest::{Client, StatusCode};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use tokio::time::sleep;
+use tokio::sync::RwLock;
+
+// -------------------------------------
+// 1) Errors
+// -------------------------------------
+
+/// Errors while talking to DDS.
+#[derive(Debug, Error)]
+pub enum DdsError {
+    /// Any reqwest/http error.
+    #[error("http error: {0}")]
+    Http(#[from] reqwest::Error),
+
+    /// Non-2xx status from DDS.
+    #[error("server returned {0}, body: {1}")]
+    BadStatus(StatusCode, String),
+
+    /// JSON (de)serialization error.
+    #[error("json error: {0}")]
+    Json(#[from] serde_json::Error),
+}
+
+// -------------------------------------
+// 2) Payload types
+// -------------------------------------
+
+/// What we send to DDS.
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct DomainServerIn {
```

Comment:
These structs are not relevant anymore. Registration will be done through websockets, not HTTP calls.